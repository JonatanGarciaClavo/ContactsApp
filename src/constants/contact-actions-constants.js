export const LOADING_CONTACT = 'LOADING_CONTACT';
export const INITIALIZE_CONTACT = 'INITIALIZE_CONTACT';
export const UPDATE_CONTACT_ATTRIBUTE = 'UPDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT_ATTRIBUTE = 'VALIDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT = 'VALIDATE_CONTACT';
export const REQUEST_CONTACT = 'REQUEST_CONTACT';
export const REQUEST_SAVE_CONTACT = 'REQUEST_SAVE_CONTACT';
