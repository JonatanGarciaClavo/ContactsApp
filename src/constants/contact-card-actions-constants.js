export const LOADING_CONTACT_CARD = 'LOADING_CONTACT_CARD';
export const INITIALIZE_CONTACT_CARD = 'INITIALIZE_CONTACT_CARD';
export const REQUEST_CONTACT_CARD = 'REQUEST_CONTACT_CARD';
export const REQUEST_DELETE_CONTACT_CARD = 'REQUEST_DELETE_CONTACT_CARD';
export const INITILIZE_CONTACT_CARD_FROM_OTHER_VIEW = 'INITILIZE_CONTACT_CARD_FROM_OTHER_VIEW';
