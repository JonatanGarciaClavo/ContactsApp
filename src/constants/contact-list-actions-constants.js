export const LOADING_CONTACT_LIST = 'LOADING_CONTACT_LIST';
export const INITIALIZE_CONTACT_LIST = 'INITIALIZE_CONTACT_LIST';
export const CHANGE_LIST_MODE = 'CHANGE_LIST_MODE';
