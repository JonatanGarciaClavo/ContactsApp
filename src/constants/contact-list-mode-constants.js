export const LIST_MODE = 'list';
export const CARD_MODE = 'card'
