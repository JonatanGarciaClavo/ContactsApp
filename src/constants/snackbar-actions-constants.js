export const CLOSE_ERROR_MESSAGE = 'CLOSE_ERROR_MESSAGE';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
