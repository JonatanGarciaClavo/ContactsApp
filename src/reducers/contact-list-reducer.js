import { Record, List } from 'immutable';
import { REQUEST_CONTACT_LIST, REQUEST_CONTACT_LIST_SUCCESS, CHANGE_LIST_MODE, RESET_CONTACT_LIST }
  from '../constants/contact-list-actions-constants';
import { SET_ERROR_MESSAGE } from '../constants/snackbar-actions-constants';
import { LIST_MODE } from '../constants/contact-list-mode-constants';

export const ContactList = new Record({
  loading: false,
  contacts: new List(),
  orderBy: 'name',
  mode: LIST_MODE,
});

const contactList = (state = new ContactList(), action) => {
  switch (action.type) {
    case (REQUEST_CONTACT_LIST):
      return state.set('loading', true);
    case (SET_ERROR_MESSAGE):
      return state.set('loading', false);
    case (REQUEST_CONTACT_LIST_SUCCESS):
      return state
        .set('contacts', action.contacts)
        .set('loading', false);
    case (CHANGE_LIST_MODE):
      if (action.mode === state.get('mode')) {
        return state;
      }
      return state.set('mode', action.mode || LIST_MODE);
    case (RESET_CONTACT_LIST):
      return new ContactList();
    default:
      return state;
  }
};

export default contactList;
