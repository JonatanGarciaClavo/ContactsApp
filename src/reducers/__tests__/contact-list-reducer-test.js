import contactList from '../contact-list-reducer';
import { initialState } from '../contact-list-reducer';
import { LOADING_CONTACT_LIST, INITIALIZE_CONTACT_LIST, CHANGE_LIST_MODE,
   } from '../../constants/contact-list-actions-constants';
import { CARD_MODE } from '../../constants/contact-list-mode-constants';

const contacts = [
  {
    id: '-1',
    email: 'test@test.com',
    imgUrl: 'http://www.test.image.com',
    name: 'Tester',
    phoneNumber: '666666666',
  },
  {
    id: '-2',
    email: 'test2@test.com',
    imgUrl: 'http://www.test2.image.com',
    name: 'Tester2',
    phoneNumber: '766666666',
  },
];

describe('Test contact list reducer', () => {
  it('returns the initialState state on an undefined state', () => {
    expect(contactList(undefined, { type: '_NULL' })).toMatchSnapshot();
  });
  it('returns the same state on an unhandled action', () => {
    expect(contactList(initialState, { type: '_NULL' })).toMatchSnapshot();
  });
  it('handles LOADING_CONTACT_LIST action', () => {
    expect(contactList(initialState, { type: LOADING_CONTACT_LIST })).toMatchSnapshot();
  });
  it('handles INITIALIZE_CONTACT_LIST action', () => {
    expect(contactList(initialState, { type: INITIALIZE_CONTACT_LIST, contacts }))
      .toMatchSnapshot();
  });
  it('handles CHANGE_LIST_MODE action', () => {
    expect(contactList(initialState, { type: CHANGE_LIST_MODE, mode: CARD_MODE }))
      .toMatchSnapshot();
  });
  it('handles CHANGE_LIST_MODE without send new mode action', () => {
    expect(contactList(initialState, { type: CHANGE_LIST_MODE }))
      .toMatchSnapshot();
  });
});
