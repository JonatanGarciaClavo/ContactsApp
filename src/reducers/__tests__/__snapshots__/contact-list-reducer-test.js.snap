exports[`Test contact list reducer handles CHANGE_LIST_MODE action 1`] = `
Object {
  "contacts": Array [],
  "loading": false,
  "mode": "card",
  "orderBy": "name",
}
`;

exports[`Test contact list reducer handles CHANGE_LIST_MODE without send new mode action 1`] = `
Object {
  "contacts": Array [],
  "loading": false,
  "mode": "list",
  "orderBy": "name",
}
`;

exports[`Test contact list reducer handles INITIALIZE_CONTACT_LIST action 1`] = `
Object {
  "contacts": Array [
    Object {
      "email": "test@test.com",
      "id": "-1",
      "imgUrl": "http://www.test.image.com",
      "name": "Tester",
      "phoneNumber": "666666666",
    },
    Object {
      "email": "test2@test.com",
      "id": "-2",
      "imgUrl": "http://www.test2.image.com",
      "name": "Tester2",
      "phoneNumber": "766666666",
    },
  ],
  "loading": false,
  "mode": "list",
  "orderBy": "name",
}
`;

exports[`Test contact list reducer handles LOADING_CONTACT_LIST action 1`] = `
Object {
  "contacts": Array [],
  "loading": true,
  "mode": "list",
  "orderBy": "name",
}
`;

exports[`Test contact list reducer returns the initialState state on an undefined state 1`] = `
Object {
  "contacts": Array [],
  "loading": false,
  "mode": "list",
  "orderBy": "name",
}
`;

exports[`Test contact list reducer returns the same state on an unhandled action 1`] = `
Object {
  "contacts": Array [],
  "loading": false,
  "mode": "list",
  "orderBy": "name",
}
`;
